File: XPile 3-17/Programs/test_const_expr.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)

MAX = 100
DOUBLE_MAX = (MAX * 2)
OFFSET = (MAX + 1)
PI = 3.14159
TAU = (PI * 2)
GREETING = ('Hello' + ' World')
FLAG = True

def main():
    print(MAX)
    print(DOUBLE_MAX)
    print(OFFSET)
    print(TAU)
    print(GREETING)
    print(FLAG)
    x = (MAX + 10)
    print(x)
    y = (DOUBLE_MAX - OFFSET)
    print(y)
    return 0


if __name__ == "__main__":
    main()

File: XPile 3-17/Programs/test_const_expr.js

const MAX = 100;
const DOUBLE_MAX = (MAX * 2);
const OFFSET = (MAX + 1);
const PI = 3.14159;
const TAU = (PI * 2);
const GREETING = ("Hello" + " World");
const FLAG = true;
function main() {
    console.log(MAX);
    console.log(DOUBLE_MAX);
    console.log(OFFSET);
    console.log(TAU);
    console.log(GREETING);
    console.log(FLAG);
    let x = (MAX + 10);
    console.log(x);
    let y = (DOUBLE_MAX - OFFSET);
    console.log(y);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
